src/coreclr/jit/valuenum.h

```diff
@@ -1167,6 +1167,16 @@ class ValueNumStore
                                       bool           encodeResultType,
                                       ValueNum       resultTypeVN);
 
+    ValueNum EvalHWIntrinsicFunTernary(var_types      type,
```

Comment:
why is it called ternary?

Reply:
There are `EvalHWIntrinsicFunUnary/Binary` for operations with 1/2 operands, this one is for 3 operands, thus ternary.